script/install.sh

```diff
@@ -407,6 +409,14 @@ parse_args() {
 
 # run_command command [compiler]
 run_command() {
+    # check if tools where installed already and update d-keyring.gpg if necessary
+    local tmp
+    tmp=$(mkdtemp)
+    if [ -f "$ROOT/d-keyring.gpg" ] && [ "$(command curl https://dlang.org/d-keyring.gpg -z "$ROOT/d-keyring.gpg" -o "$tmp/d-keyring.gpg" -s -L -w "%{http_code}")" == "200" ] ; then
```

Comment:
This is the real problem:
1) This probably needs retry logic
2) Probably needs the fallback mirrors
3) Should verify the authenticity
4) Shouldn't re-download the file(s) if they haven't changed
5) Needs to update `install.sh` as well (the script changes a lot more often than the keyring)
6) Needs to provide the user an option to opt-out
7) Should probably use "Last-Modified", "ETag" or similar to avoid unnecessary downloads
8) Auto-updates need to be documented in help page, documentation and changelog (there are serious security concerns with it)

Reply:
this all together is far beyond my bash foo and also the time I can / want spend on this topic, sorry...
nevertheless, some comments from here...

4. => make's sense but I couldn't find a (smart & easy) way how to check if the file was changed without downloading it. actually I did read plenty of "curl way" post / blogs, i.e. https://blog.yjl.im/2012/03/downloading-only-when-modified-using.html and https://www.thegeekstuff.com/2012/04/curl-examples/ but at the end it seems like curl ALLWAYS downloads the file to be able to check for a change... and as it's downloaded anyhow, I decided to implement it as it is

5. => well, if the script itself should also be updated, neither this change nor the "update" option of the script makes any sense... because in other words you ask to make the already existing"update" option an always applied defauld (from my point of view, I wouldn't like this kind of change... I want to know and control when and why a tool gets updated)

7. => see my comments on topic nr4